js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+        const status = marketStatus === 'success' ? 'ok' : marketStatus;
+        return {
+            'info': transaction,
+            'id': this.safeString (transaction, 'txid'), // Exchange doesn't provide its own id
+            'txid': this.safeString (transaction, 'txid'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'address': this.safeString (transaction, 'address'),
+            'type': type,
+            'amount': this.asFloat (this.safeString (transaction, 'amount')),
+            'currency': this.commonCurrencyCode (this.safeString (transaction, 'coin')),
+            'status': status,
+            'updated': undefined,
+            'tag': undefined,
+            'fee': undefined,
+        };
+    }
+
+    async fetchDeposits (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let res = await this.privatePost ({ 'method': 'getmydeposithistory' });
+        const deposits = this.parseJson (res);
+        // Some deposits seem to have no data associated
+        const validDeposits = [];
+        for (let i = 0; i < deposits.length; i++) {
+            const deposit = deposits[i];
+            if (deposit.txid !== undefined) {
+                validDeposits.push (deposit);
+            }
+        }
+        const result = this.parseTransactions (validDeposits, code, since, limit, { 'type': 'deposit' });
+        return result;
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let res = await this.privatePost ({ 'method': 'getmywithdrawhistory' });
+        const withdrawals = this.parseJson (res);
+        // Some withdrawals seem to have no data associated
+        const validWithdrawals = [];
+        for (let i = 0; i < withdrawals.length; i++) {
+            const withdrawal = withdrawals[i];
+            if (withdrawal.txid !== undefined) {
+                validWithdrawals.push (withdrawal);
+            }
+        }
+        const result = this.parseTransactions (validWithdrawals, code, since, limit, { 'type': 'withdrawal' });
+        return result;
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        if (code === undefined || amount === undefined || address === undefined) {
+            throw new ArgumentsRequired (this.id + ' withdraw () requires a code, amount and address argument');
+        }
+        if (address.indexOf ('0x') === 0) {
+            address = address.substr (2);
+        }
+        const withdrawRequest = {
+            'method': 'withdraw',
+            'coin': this.currencyId (code),
+            'address': address,
+            'amount': amount,
+        };
+        const result = await this.privatePost (withdrawRequest);
+        return { 'info': result };
+    }
+
+    nonce () {
+        return this.seconds ();
```

Comment:
Can this support milliseconds, or even better, microseconds?  I prefer the biggest possible one in case we need to make a lot of requests

Reply:
Sure--I'm totally unsure what to do here to specify that the exchange ignores nonce. I guess I'll add a comment here.

Reply:
ohhhh right, well maybe they'll eventually fix it.  See if mircoseconds works and just put that in